File: sistema-penal/src/models/Funcionario.ts
import { Pessoa } from "./abstract/Pessoa";
import type { Role } from "./enum/Role";

export class Funcionario extends Pessoa {
  email: string;
  papel: Role;
  senha: string;

  constructor(
    nome: string,
    dataNascimento: Date,
    cpf: string,
    email: string,
    papel: Role,
    senha: string,
    id?: string,
  ) {
    super(nome, dataNascimento, cpf, id);
    this.email = email;
    this.papel = papel;
    this.senha = senha;
  }
}

File: api/SistemaPenal/Entities/Funcionario.cs
using SistemaPenal.Entities.Abstractions;
using SistemaPenal.Enums;

namespace SistemaPenal.Entities;

public class Funcionario : Pessoa
{
    public required string Email { get; set; }
    public Role Papel { get; set; }
    public required string Senha { get; set; }
}